hbase-client/src/main/java/org/apache/hadoop/hbase/client/HRegionLocator.java

```diff
@@ -122,6 +135,37 @@ public boolean visitInternal(Result result) throws IOException {
       }
     };
     MetaTableAccessor.scanMetaForTableRegions(connection, visitor, tableName);
-    return regions;
+    return consolidate(regions);
+  }
+
+  private static RegionLocations consolidate(final List<RegionLocations> locations) {
+    final HRegionLocation[] consolidated = locations.stream()
+      .filter(Objects::nonNull)
+      .flatMap(locs -> Arrays.stream(locs.getRegionLocations()))
+      .filter(Objects::nonNull)
+      .toArray(HRegionLocation[]::new);
```

Comment:
Should this deduplicate any`RegionLocations`? I guess from above, the answer is "no" (we're just massaging the results we got out of meta) and we should just trust that meta is reasonable.

Reply:
Disregard. Per Huaxiang's comments, I'm reverting this method.